src/components/RuleBuilder/Rule.jsx

```diff
@@ -44,29 +44,38 @@ export const Rule = ({
   const type = useMemo(() => rule.type, [rule])
 
   const operatorName = useMemo(() => `${name}.${type}Operator`, [name, type])
-  const [{ value: operatorValue }] = useField(operatorName)
-
+  const [{ value: operatorValue }, , { setValue: setOperatorValue }] =
+    useField(operatorName)
   const operator = useMemo(() => options[type].operator, [options, type])
   const value = useMemo(
     () => options[type]?.operatorValue?.[operatorValue] ?? options[type].value,
     [options, type, operatorValue]
   )
 
-  const typePlaceholder = useMemo(() => options[type].placeholder, [
-    options,
-    type
-  ])
+  const typePlaceholder = useMemo(
+    () => options[type].placeholder,
+    [options, type]
+  )
 
   const valuePlaceholder = useMemo(() => value.placeholder, [value])
 
   const isSuffixType = useMemo(() => value.suffix, [value])
   const [, , { setValue: setRuleValue }] = useField(`${name}.${type}Value`)
 
   const onChangeRuleValue = useClearField([setRuleValue])
+  const [shouldReset, setShouldReset] = useState(false)
+
+  const handleOperatorChange = newValue => {
+    setOperatorValue(newValue)
+    setShouldReset(true)
```

Comment:
should the order be inversed ?
1. mark should reset true
2. change the value -> this triggers the effect

Reply:
I don't think so. The efect depends on the `shouldReset` state, not on the actual `operatorValue`. The setting of the `shouldReset` state is triggering the effect.

Reply:
There are two cases when the `operatorValue` can change.

1. The RuleBuilder is rendered initially, then the `operatorValue` receives a new value (from import)
2. The user selects a new `operatorValue` from the dropdown. ~ This is the case when we want to reset the state.

Reply:
f0f1c95d646aacf342452fcf346c45990638b44b